src/chainstate/stacks/db/transactions.rs

```diff
@@ -3555,6 +3555,114 @@ pub mod test {
+
+        let contract = "
+(define-constant owner 'ST3X2W2SH9XQZRHHYJ21KWGTT1N6WX3D48K1NSTPE)
+(define-fungible-token connect-token)
+(begin (ft-mint? connect-token u100000000 owner))
+(define-public (transfer (recipient principal) (amount uint))
+  (ok (ft-transfer? connect-token amount tx-sender recipient)))
+"
+        .to_string();
+
+        let contract_name = ContractName::try_from("hello-world").unwrap();
+
+        let recv_addr = StacksAddress::from_public_keys(
+            C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
+            &AddressHashMode::SerializeP2PKH,
+            1,
+            &vec![StacksPublicKey::from_private(&privk_recipient)],
+        )
+        .unwrap();
+        let recv_principal = recv_addr.to_account_principal();
+        let contract_id = QualifiedContractIdentifier::new(
+            StandardPrincipalData::from(addr_publisher.clone()),
+            contract_name.clone(),
+        );
+        let _contract_principal = PrincipalData::Contract(contract_id.clone());
+
+        let asset_info = AssetInfo {
+            contract_address: addr_publisher.clone(),
+            contract_name: contract_name.clone(),
+            asset_name: ClarityName::try_from("connect-token").unwrap(),
+        };
+
+        let mut tx_contract = StacksTransaction::new(
+            TransactionVersion::Testnet,
+            auth_origin.clone(),
+            TransactionPayload::new_smart_contract(&"hello-world".to_string(), &contract).unwrap(),
+        );
+
+        tx_contract.chain_id = 0x80000000;
+        tx_contract.set_fee_rate(0);
+
+        let mut signer = StacksTransactionSigner::new(&tx_contract);
+        signer.sign_origin(&privk_origin).unwrap();
+
+        let signed_contract_tx = signer.get_tx().unwrap();
+
+        let mut tx_contract_call = StacksTransaction::new(
+            TransactionVersion::Testnet,
+            auth_origin.clone(),
+            TransactionPayload::new_contract_call(
+                addr_publisher.clone(),
+                "hello-world",
+                "transfer",
+                vec![Value::Principal(recv_principal.clone()), Value::UInt(10)],
+            )
+            .unwrap(),
+        );
+
+        tx_contract_call.chain_id = 0x80000000;
+        tx_contract_call.set_fee_rate(0);
+        tx_contract_call.set_origin_nonce(1);
+
+        tx_contract_call.post_condition_mode = TransactionPostConditionMode::Deny;
+        tx_contract_call.add_post_condition(TransactionPostCondition::Fungible(
+            PostConditionPrincipal::Origin,
+            asset_info.clone(),
+            FungibleConditionCode::SentEq,
+            10,
+        ));
+
+        let mut signer = StacksTransactionSigner::new(&tx_contract_call);
+        signer.sign_origin(&privk_origin).unwrap();
+        let contract_call_tx = signer.get_tx().unwrap();
+
+        let mut chainstate =
+            instantiate_chainstate(false, 0x80000000, "process-post-conditions-tokens-deny");
```

Comment:
Might need to make this test name `process-post-conditions-tokens-deny-2097` so it doesn't conflict with any other on-disk chainstate.

Reply:
Resolved in ff936105070b3091fb599533b45206b3e88cd357